src/main/java/seedu/siasa/logic/commands/contact/EditContactCommand.java

```diff
@@ -72,22 +77,50 @@ public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Contact> lastShownList = model.getFilteredContactList();
 
+        // Check for validity of provided index
         if (index.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_CONTACT_DISPLAYED_INDEX);
         }
 
         Contact contactToEdit = lastShownList.get(index.getZeroBased());
         Contact editedContact = createEditedContact(contactToEdit, editContactDescriptor);
 
+        // Check if contact is edited to be a duplicate contact
         if (!contactToEdit.isSameContact(editedContact) && model.hasContact(editedContact)) {
             throw new CommandException(MESSAGE_DUPLICATE_CONTACT);
         }
 
+        // Update all the associated policies
+        ArrayList<Pair<Policy, Policy>> policiesToBeUpdated = new ArrayList<>();
+        model.updateFilteredPolicyList(new PolicyIsOwnedByPredicate(contactToEdit));
+        List<Policy> contactPolicies = model.getFilteredPolicyList();
```

Comment:
Hmm my understanding is that filteredpolicylist is bounded to the UI and is meant to change what is displayed on the UI. You resetted it at the end but this could be a source of unexpected behaviour due to the coupling with UI. Would it be better if we created a method in model (and hence modelmanager and siasa) that returns the contact's Policies without modifying the filteredpolicylist?

Reply:
Will do

Reply:
Updated method to not use filters as of commit bd681b38a1eef29ad0ac3b05f3dd32862c5d4157bd681b38a1eef29ad0ac3b05f3dd32862c5d4157